Unused parameter 'deterministic'

<b>OS:</b>
Ubuntu 16.04 (in Docker)

<b>Cockatrice version:</b>
latest (hash is useless b/c of unrelated local commits)

___
<br>


When building Cockatrice in a Docker container after configuring with `cmake .. -DWITH_SERVER=1 -DTEST=1 -DCMAKE_BUILD_TYPE=Debug`, I get the following error:

```
Scanning dependencies of target cockatrice_protocol
[ 31%] Building CXX object common/pb/CMakeFiles/cockatrice_protocol.dir/admin_commands.pb.cc.o
/builds/cockatrice/cockatrice/build/common/pb/admin_commands.pb.cc: In member function 'virtual google::protobuf::uint8* AdminCommand::InternalSerializeWithCachedSizesToArray(bool, google::protobuf::uint8*) const':
/builds/cockatrice/cockatrice/build/common/pb/admin_commands.pb.cc:367:10: error: unused parameter 'deterministic' [-Werror=unused-parameter]
     bool deterministic, ::google::protobuf::uint8* target) const {
          ^~~~~~~~~~~~~
/builds/cockatrice/cockatrice/build/common/pb/admin_commands.pb.cc: In member function 'virtual bool AdminCommand::IsInitialized() const':
/builds/cockatrice/cockatrice/build/common/pb/admin_commands.pb.cc:443:3: error: this 'if' clause does not guard... [-Werror=misleading-indentation]
   if (!_extensions_.IsInitialized()) return false;  return true;
   ^~
/builds/cockatrice/cockatrice/build/common/pb/admin_commands.pb.cc:443:53: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the 'if'
   if (!_extensions_.IsInitialized()) return false;  return true;
                                                     ^~~~~~
/builds/cockatrice/cockatrice/build/common/pb/admin_commands.pb.cc: In member function 'virtual google::protobuf::uint8* Command_UpdateServerMessage::InternalSerializeWithCachedSizesToArray(bool, google::protobuf::uint8*) const':
/builds/cockatrice/cockatrice/build/common/pb/admin_commands.pb.cc:583:10: error: unused parameter 'deterministic' [-Werror=unused-parameter]
     bool deterministic, ::google::protobuf::uint8* target) const {
          ^~~~~~~~~~~~~
/builds/cockatrice/cockatrice/build/common/pb/admin_commands.pb.cc: In member function 'virtual google::protobuf::uint8* Command_ShutdownServer::InternalSerializeWithCachedSizesToArray(bool, google::protobuf::uint8*) const':
/builds/cockatrice/cockatrice/build/common/pb/admin_commands.pb.cc:853:10: error: unused parameter 'deterministic' [-Werror=unused-parameter]
     bool deterministic, ::google::protobuf::uint8* target) const {
          ^~~~~~~~~~~~~
/builds/cockatrice/cockatrice/build/common/pb/admin_commands.pb.cc: In member function 'virtual google::protobuf::uint8* Command_ReloadConfig::InternalSerializeWithCachedSizesToArray(bool, google::protobuf::uint8*) const':
/builds/cockatrice/cockatrice/build/common/pb/admin_commands.pb.cc:1182:10: error: unused parameter 'deterministic' [-Werror=unused-parameter]
     bool deterministic, ::google::protobuf::uint8* target) const {
          ^~~~~~~~~~~~~
/builds/cockatrice/cockatrice/build/common/pb/admin_commands.pb.cc: In member function 'virtual google::protobuf::uint8* Command_AdjustMod::InternalSerializeWithCachedSizesToArray(bool, google::protobuf::uint8*) const':
/builds/cockatrice/cockatrice/build/common/pb/admin_commands.pb.cc:1452:10: error: unused parameter 'deterministic' [-Werror=unused-parameter]
     bool deterministic, ::google::protobuf::uint8* target) const {
          ^~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
common/pb/CMakeFiles/cockatrice_protocol.dir/build.make:1278: recipe for target 'common/pb/CMakeFiles/cockatrice_protocol.dir/admin_commands.pb.cc.o' failed
make[2]: *** [common/pb/CMakeFiles/cockatrice_protocol.dir/admin_commands.pb.cc.o] Error 1
CMakeFiles/Makefile2:1073: recipe for target 'common/pb/CMakeFiles/cockatrice_protocol.dir/all' failed
make[1]: *** [common/pb/CMakeFiles/cockatrice_protocol.dir/all] Error 2
```

This doesn't happen when building in Release mode in an identical container.

Full build log: https://gitlab.tetrarch.co/cockatrice/cockatrice/builds/714

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -92,6 +92,9 @@ elseif(WIN32)
     set(CMAKE_INSTALL_PREFIX ${CMAKE_BINARY_DIR}/release)
 endif()
 
+# Treat warnings as errors (Debug builds only)
+option(WARNING_AS_ERROR "Treat warnings as errors in debug builds" ON)
+
 # Define proper compilation flags
 IF(MSVC)
     # Visual Studio:
@@ -104,7 +107,11 @@ ELSEIF (CMAKE_COMPILER_IS_GNUCXX)
     include(CheckCXXCompilerFlag)
 
     set(CMAKE_CXX_FLAGS_RELEASE "-s -O2")
-    set(CMAKE_CXX_FLAGS_DEBUG "-ggdb -O0 -Wall -Wextra -Werror")
+    if(WARNING_AS_ERROR)
+        set(CMAKE_CXX_FLAGS_DEBUG "-ggdb -O0 -Wall -Wextra -Werror")
+    else()
+        set(CMAKE_CXX_FLAGS_DEBUG "-ggdb -O0 -Wall -Wextra")
+    endif()
 
     set(ADDITIONAL_DEBUG_FLAGS -Wcast-align -Wmissing-declarations -Wno-long-long -Wno-error=extra -Wno-error=delete-non-virtual-dtor -Wno-error=sign-compare -Wno-error=missing-declarations)
 

--- a/README.md
+++ b/README.md
@@ -70,7 +70,7 @@ Cockatrice uses the [Google Developer Documentation Style Guide](https://develop
 # Translations [![Cockatrice on Transiflex](https://tx-assets.scdn5.secure.raxcdn.com/static/charts/images/tx-logo-micro.c5603f91c780.png)](https://www.transifex.com/projects/p/cockatrice/)
 
 Cockatrice uses Transifex for translations. You can help us bring Cockatrice and Oracle to your language or just edit single wordings right from within your browser by visiting our [Transifex project page](https://www.transifex.com/projects/p/cockatrice/).<br>
-   
+
 | Cockatrice | Oracle |
 |:-:|:-:|
 | [![Cockatrice Translation Status](https://www.transifex.com/projects/p/cockatrice/resource/cockatrice/chart/image_png/)](https://www.transifex.com/projects/p/cockatrice/) | [![Oracle Translation Status](https://www.transifex.com/projects/p/cockatrice/resource/oracle/chart/image_png/)](https://www.transifex.com/projects/p/cockatrice/) |
@@ -83,7 +83,7 @@ Check out our [Translator FAQ](https://github.com/Cockatrice/Cockatrice/wiki/Tra
 **Detailed compiling instructions are on the Cockatrice wiki under [Compiling Cockatrice](https://github.com/Cockatrice/Cockatrice/wiki/Compiling-Cockatrice)**
 
 Dependencies:
-- [Qt](https://www.qt.io/developers/) 
+- [Qt](https://www.qt.io/developers/)
 - [protobuf](https://github.com/google/protobuf)
 - [CMake](https://www.cmake.org/)
 
@@ -96,7 +96,7 @@ To compile:
     cd build
     cmake ..
     make
-    
+
 You can then run
 
     make install
@@ -113,14 +113,15 @@ The following flags can be passed to `cmake`:
 - `-DWITH_CLIENT=0` Whether to build the client (default 1 = yes).
 - `-DWITH_ORACLE=0` Whether to build oracle (default 1 = yes).
 - `-DCMAKE_BUILD_TYPE=Debug` Compile in debug mode. Enables extra logging output, debug symbols, and much more verbose compiler warnings (default `Release`).
+- `-DWARNING_AS_ERROR=0` Whether to treat compilation warnings as errors in debug mode (default 1 = yes).
 - `-DUPDATE_TRANSLATIONS=1` Configure `make` to update the translation .ts files for new strings in the source code. Note: Running `make clean` will remove the .ts files (default 0 = no).
 - `-DTEST=1` Enable regression tests (default 0 = no). Note: needs googletest, will be downloaded on the fly if unavailable. To run tests: ```make test```.
 
 
 # Run
 
-`Cockatrice` is the game client    
-`Oracle` fetches card data    
+`Cockatrice` is the game client
+`Oracle` fetches card data
 `Servatrice` is the server<br>
 
 

